gapic/utils/case.py

```diff
@@ -45,3 +45,19 @@ def to_snake_case(s: str) -> str:
 
     # Done; return the camel-cased string.
     return s.lower()
+
+
+def to_camel_case(s: str) -> str:
+    '''Convert any string to camel case.
+
+    This is provided to templates as the ``camel_case`` filter.
+
+    Args:
+        s (str): The input string, provided in any sane case system
+
+    Returns:
+        str: The string in lower camel case.
+    '''
+
+    items = re.split(r'[_-]', to_snake_case(s))
```

Comment:
I'm surprised by the hyphen `-`. It almost seems like that should be handled by `to_snake_case`, though that's outside the scope of this PR. Does it come up in practice, that an input to this function has hyphens/kebab-case?

Opinion, @software-dov ?

Reply:
It actually might handle it: I didn't check. As far as use, it doesn't get used but I wouldn't want a future user to attempt o try it and have it fail.

Reply:
My vote: write a test for `to_snake_case`'s behavior, add the functionality if the test fails, just split on '_'.

Reply:
If `to_snake_case` does not handle it, could adding the functionality break existing clients? is there an easy way to check? (This is why I was suggesting "outside the scope of this PR")

Reply:
I doubt it would break anything since it's additive but this is why I didn't change `to_snake_case`'s behavior

Reply:
But the generator has already produced libraries that were published, so if one of those had a hyphen somewhere, it made it through `to_snake_case` and changing that implementation could lead to an incompatible surface.